it/v1/endpoints/CreateMarriageAllowanceControllerISpec.scala

```diff
@@ -0,0 +1,275 @@
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.Status._
+import play.api.libs.json.{ JsValue, Json }
+import play.api.libs.ws.{ WSRequest, WSResponse }
+import support.IntegrationBaseSpec
+import v1.models.errors._
+import v1.stubs.{ AuditStub, AuthStub, DesStub, MtdIdLookupStub }
+
+class CreateMarriageAllowanceControllerISpec extends IntegrationBaseSpec {
+
+  private trait Test {
+
+    val nino1: String         = "AA123456A"
+    val nino2: String         = "BB123456B"
+    val invalidNino: String   = "BB123456Br"
+    val correlationId: String = "X-123"
+
+    val requestBodyJson: JsValue = Json.parse(
+      s"""
+        |{
+        |  "spouseOrCivilPartnerNino": "$nino2",
+        |  "spouseOrCivilPartnerFirstName": "John",
+        |  "spouseOrCivilPartnerSurname": "Smith",
+        |  "spouseOrCivilPartnerDateOfBirth": "1986-04-06"
+        |}
+      """.stripMargin
+    )
+
+    def uri: String = s"/marriage-allowance/$nino1"
+
+    def desUri: String = s"/income-tax/marriage-allowance/claim/NINO/$nino1"
+
+    def setupStubs(): StubMapping
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+  }
+
+  "Calling the 'create marriage allowence' endpoint" should {
+    "return a 201 status code" when {
+      "any valid request is made" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino1)
+          DesStub.onSuccess(DesStub.POST, desUri, NO_CONTENT)
+        }
+
+        val response: WSResponse = await(request().post(requestBodyJson))
+        response.status shouldBe CREATED
+        response.body[String] shouldBe ""
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return a 400 with multiple errors" when {
+      "all field value validations fail on the request body" in new Test {
+
+        val allInvalidValueRequestBodyJson: JsValue = Json.parse(
+          s"""
+             |{
+             |  "spouseOrCivilPartnerNino": "$invalidNino",
+             |  "spouseOrCivilPartnerFirstName": "Johny1n4|y8nx34tij8",
+             |  "spouseOrCivilPartnerSurname": "Smith47cyw-i|teqytya",
+             |  "spouseOrCivilPartnerDateOfBirth": "1943286-04-06"
+             |}
+          """.stripMargin
+        )
+
+        val allInvalidValueRequestError: List[MtdError] = List(
+          BadRequestError.copy(code = PartnerFirstNameFormatError.code, message = PartnerFirstNameFormatError.message),
+          BadRequestError.copy(code = PartnerNinoFormatError.code, message = PartnerNinoFormatError.message),
+          BadRequestError.copy(code = PartnerDoBFormatError.code, message = PartnerDoBFormatError.message),
+          BadRequestError.copy(code = PartnerSurnameFormatError.code, message = PartnerSurnameFormatError.message)
+        )
```

Comment:
This is a really weird way to do this. Why not just
```suggestion
        val allInvalidValueRequestError: List[MtdError] = List(
          PartnerFirstNameFormatError
          PartnerNinoFormatError,
          PartnerDoBFormatError,
          PartnerSurnameFormatError
        )
```